backend/gzchain/chain.go

```diff
@@ -229,14 +299,94 @@ func (g *gnoZenaoChain) CreateEvent(evtID string, organizersIDs []string, gateke
 	return nil
 }
 
+// GetEvent implements ZenaoChain.
+func (g *gnoZenaoChain) GetEvent(eventRealmID string) (*zenaov1.EventInfo, error) {
+	g, span := g.trace("gzchain.GetEvent")
+	defer span.End()
+
+	raw, err := checkQueryErr(g.client.QEval(eventRealmID, "event.Info().ToJSON().String()"))
+	if err != nil {
+		return nil, err
+	}
+	parsedRaw, err := parseQEvalStringResponseData(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	var res zenaov1.EventInfo
+	if err := protojson.Unmarshal([]byte(parsedRaw), &res); err != nil {
+		return nil, err
+	}
+
+	return &res, nil
+}
+
+// GetEventUsersByRole implements ZenaoChain.
+func (g *gnoZenaoChain) GetEventUsersByRole(eventRealmID string, role string) ([]string, error) {
+	g, span := g.trace("gzchain.GetEventUsersByRole")
+	defer span.End()
+
+	raw, err := checkQueryErr(g.client.QEval(eventRealmID, "event.GetUsersWithRoleJSON(\""+role+"\")"))
+	if err != nil {
+		return nil, err
+	}
+	parsedRaw, err := parseQEvalStringResponseData(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	var userRealmIDs []string
+	if err := json.Unmarshal([]byte(parsedRaw), &userRealmIDs); err != nil {
+		return nil, err
+	}
+
+	return userRealmIDs, nil
+}
+
+// GetEventCommunity implements ZenaoChain.
+func (g *gnoZenaoChain) GetEventCommunity(eventRealmID string) (*zenaov1.CommunityInfo, error) {
+	g, span := g.trace("gzchain.GetEventCommunity")
+	defer span.End()
+
+	raw, err := checkQueryErr(g.client.QEval(g.communitiesIndexPkgPath, "communitiesToJSON(listCommunitiesByEvent(\""+eventRealmID+"\", 1, 0))"))
+	if err != nil {
+		return nil, err
+	}
+	parsedRaw, err := parseQEvalStringResponseData(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	// protojson expects a single protobuf message so i parse them into a list of raw json msg first
+	var rawList []json.RawMessage
+	if err := json.Unmarshal([]byte(parsedRaw), &rawList); err != nil {
+		return nil, err
+	}
+
+	list := make([]*zenaov1.CommunityInfo, 0, len(rawList))
+	for _, elem := range rawList {
+		var info zenaov1.CommunityInfo
+		if err := protojson.Unmarshal(elem, &info); err != nil {
+			return nil, err
+		}
+		list = append(list, &info)
+	}
+
+	if len(list) == 0 {
+		return nil, nil
+	}
+	if len(list) > 1 {
```

Comment:
TODO: either limit of 1 community on chain or handle multiple community but align back & chain